### ISSUE TITLE
[bench request]: decider-4b v2 (Mapika/decider-4b, 4.2B, TypeSafe wire format, self-hosted)

### ISSUE THREAD

**Issue by Mapika** [fstandhartinger/jevbench#79]
**Package:** [`decider-ai` 1.2.2](https://pypi.org/project/decider-ai/1.2.2/) on PyPI, built from https://github.com/Mapika/decider
at [`abadc94`](https://github.com/Mapika/decider/tree/v1.2.2) (tag `v1.2.2`) · weights https://huggingface.co/Mapika/decider-4b at
[`7ab294c`](https://huggingface.co/Mapika/decider-4b/tree/7ab294cbdf6be6ac17fc818c10cdead744393d92)
(`7ab294cbdf6be6ac17fc818c10cdead744393d92`, the current `main`; `decider_config.json` there says `"version": "4b-v2"`,
`"temperature": 1.935`)

Please add decider-4b v2 as its own row. It is the 4B dense member of the decider family; decider-2b (#2) and decider-35b-a3b
(#4) are already on the board, and this request does not replace or change either of them. decider-4b v1 was released on
2026-09-22 but was never requested or listed, so there is no earlier 4B row.

**What it is.** A decision model that does not generate text. The server renders the state, the question and its options into
one prompt, runs one forward pass, and reads the logits of the option-letter tokens at the answer slot; a softmax at one
global temperature (1.935) gives the probability of every option. No tokens are decoded and `usage.output_tokens` is always 0.
Every question is scored in its own row. Base: Qwen/Qwen3.5-4B-Base (4.2B parameters), Apache-2.0 weights and code.

**What we measured with JevBench's CLI at `2fa63fa` on the public set (231 items):**

| | NVIDIA B300, bf16 |
|---|---|
| correct | 193/231 |
| easy / standard / hard | 48/48 · 71/72 · 74/111 |
| answered and valid | 231/231 (strict-valid 231, renormalized 0) |
| ECE | 0.031 (hard tier alone, same binning: 0.085) |
| latency p50 / p95 | 13 ms / 67 ms |
| mean input tokens | 629 |

Latency is raw: one request at a time, client and server on the same machine, no other load on the GPU, package defaults
(no FP8, no `torch.compile`; CUDA graphs are captured at start-up, which takes about 50 s).

**Identity check:** a correct setup reproduces 193/231 with that tier split (48 · 71 · 74) and model name `decider-4b-v2`.

Our model card gives the hard tier as 0.676 (75/111). That figure came from an earlier in-process read without CUDA graphs.
The two reads differ on one item, `hard-sol-b-long_policy-02`, where the top two options are within 0.03 of each other
(earlier read: 0.380 and 0.380, resolved to the gold; this read: 0.387 for `privacy_only` against 0.363 for the gold).
The HTTP server above is the configuration we ask you to run, and 74/111 is its number.

**To run it.** The service speaks your wire format (`POST /v1/systemone`), so the existing `typesafe` adapter reaches it with
no new code:

```sh
python3.12 -m venv .venv && . .venv/bin/activate
pip install "decider-ai[serve]==1.2.2"          # pulls torch 2.14, transformers 5.17, flash-linear-attention
CUDA_VISIBLE_DEVICES=0 DECIDER_MODEL=Mapika/decider-4b \
    python -m uvicorn decider.serve:app --host 127.0.0.1 --port 8000
# first start downloads 8.4 GB; the server prints "[serve] ready {"model": "decider-4b-v2", ... "temperature": 1.935 ...}"

python3 -m jevbench.cli run \
    --tasks datasets/public/easy.jsonl,datasets/public/original.jsonl,datasets/public/hard.jsonl \
    --adapter typesafe --endpoint http://127.0.0.1:8000 --model decider-4b-v2 \
    --key-env "" --results out.jsonl
```

To pin the weights to the revision above instead of `main`, download them first
(`huggingface-cli download Mapika/decider-4b --revision 7ab294cbdf6be6ac17fc818c10cdead744393d92 --local-dir decider-4b-v2`)
and set `DECIDER_MODEL=./decider-4b-v2`; the server reads `decider_config.json` from that folder. Our run used the Hub id while
`main` was `7ab294c`.

The package turns off PyTorch's cuDNN SDPA backend itself at engine start (it gives wrong results for masked attention on
Blackwell with torch 2.14); nothing needs to be set for that. No hosted endpoint exists, so the run is self-hosted on your
GPU and the row needs no API flag.

decider-ai 1.3.0 was released after this run. It changes only the `confidence` field of `/v1/systemone` answers (now TypeSafe's
definition) and accepts noul questions without `instructions`; the probabilities your adapter reads are the same. The numbers
above were taken with 1.2.2, which is the version pinned in the commands.

**Disclosures**

- **Base and training.** decider-4b v2 is decider-4b v1 plus a merged LoRA. decider-4b v1 is Qwen/Qwen3.5-4B-Base after one
  supervised pass over our public decision mixture (mixture v2). The v2 LoRA (rank 64, alpha 128, attention and MLP weights,
  2 epochs, 29,356 rows) was trained on:
  - 8,000 rows from ten programmatic decision families that we generated to mirror the published names of your ten sealed
    families in `docs/METHOD-v1.4.md` (temporal and numeric decisions, subtle answer judgment, long policies, multi-hop lookup,
    abstention, probability, constrained tradeoffs, safety judgment, paraphrase sensitivity, adversarial traps). Answers are
    computed by code from the generated facts. The generators were written from those names and that one-line description,
    without reading any JevBench item. During a later data-format check, one public hard probability item's provenance field
    was printed; the generators were already written at that point and it was not used for anything.
  - 11,356 rows of questions written by Qwen3.6-27B with thinking on over business documents it also wrote, kept only when two
    independent Qwen3.6-27B answers agreed with the writer's answer.
  - 3,300 rows from human-labelled public datasets (MMLU, ARC, CommonsenseQA, BoolQ, MNLI, SNLI, Banking77, RACE,
    OpenBookQA, LogiQA 2, MedQA, WinoGrande; train splits).
  - 6,700 rows of replay from v1's mixture v2.
- **No JevBench item, public or sealed, was used for training, data selection, checkpoint selection, the temperature or the
  prompt layout.** We have never seen the sealed items. Training rows were checked for exact state-plus-question overlap
  against our own evaluation files; that check did not cover JevBench files, because no JevBench text was an input to any
  training source.
- **When we read the public items.** The checkpoint was picked from ten candidates by a selection score on our own sets. The
  public items were then read once, for this checkpoint only (at that time at temperature 1.719). Our own release rule for that
  run did not pass (the candidate tied an earlier candidate on our hard sets instead of beating it); the decision to release it
  anyway was made after that public readout and after further measurements on our own sets. The public items were read a
  second time for the identity check above. Earlier models of the same line (decider-4b v1 and one earlier LoRA candidate) were
  also read once each on the public items.
- **Temperature.** One global temperature, 1.935, fitted by NLL on our own in-task rows (61 tasks of our public regression set,
  102,804 rows). It applies to every question type. No per-type or per-tier temperature.
- **Context limit.** The state is limited to 32,768 tokens (`DECIDER_MAX_STATE_TOKENS`). A longer state is **not** refused: the
  tokens of `"Context:\n" + state` are cut after the first 32,768, so the end of the state is dropped, and the question is
  answered on the rest. A request is refused with HTTP 413 only when one scoring row (truncated state plus question and options)
  exceeds 36,864 tokens, or the whole request exceeds 1,048,576 tokens. The longest public item here is 3,911 input tokens, so
  nothing was truncated in the run above.
- **Prompt layout.** Plain text, state first: `Context:\n<state>`, then `\n\nQuestion: <instructions>\nOptions:` with one
  line per option, then `\nAnswer: (`. Options are listed in the order `criteria` gives them, with no shuffling; each is shown as
  `(A) name` or, when the criterion has a description, `(A) name: description`. More than ten options switch to single-token
  labels A..Z, AA, AB, ... (up to 255). A structured state is rendered with `json.dumps(..., ensure_ascii=False)`, with an
  `_index` field added to the elements of arrays of 8 or more elements. `noul` is asked as two options, `(A) no` and `(B) yes`
  (with their descriptions when given), and `noul` returns P(yes).
- **Score questions.** Each level is asked as its own yes/no row ("Proposed answer: <level text> / Does the proposed answer
  fit?", the level number removed), and the P(yes) values are normalized over the levels into the returned distribution. The
  answer also carries `level_fit` and `fit_mass` beside the fields your adapter reads; ignoring them changes nothing.
- **Order sensitivity.** The letter is the option's position, so a different option order is a different reading. Your adapter
  passes `criteria` through unchanged, which is what the numbers above were taken with.
- **Price basis.** Self-hosted bf16, 8.4 GB of weights, one GPU. Input tokens only; nothing is generated. We propose your hosted
  price for the 4B dense size class, labelled as an estimate. Mean input tokens on
  the public set: 629.
- **Hardware.** Measured on one NVIDIA B300 only. We have not measured it on a smaller card.

**Row details**

- **repo:** https://github.com/Mapika/decider (package `decider-ai` 1.2.2 on PyPI, tag `v1.2.2`)
- **weights:** https://huggingface.co/Mapika/decider-4b at `7ab294cbdf6be6ac17fc818c10cdead744393d92`
- **author:** Mapika
- **underlying:** `Qwen/Qwen3.5-4B-Base` (Alibaba Cloud), supervised fine-tuning (v1), then a LoRA merged into the weights (v2).
- **licence:** Apache-2.0 for the package and the weights.

The sealed items are yours to run. We have never seen them.

**Comment by fstandhartinger** [comment]
Thanks — this is logged for reachability checks and the JevBench measurement queue. We’ll only make benchmark claims after our own run.

- Harold
